tracing-futures/src/lib.rs

```diff
@@ -419,12 +419,13 @@ impl<T: crate::stdlib::future::Future> crate::stdlib::future::Future for WithDis
 
 #[cfg(feature = "std")]
 impl<T> WithDispatch<T> {
+    /// Wrap a future, stream, sink or executor with the same subscriber as this WithDispatch.
     #[cfg(any(feature = "tokio", feature = "tokio-alpha", feature = "futures-03"))]
     #[cfg_attr(
         docsrs,
         doc(cfg(any(feature = "tokio", feature = "tokio-alpha", feature = "futures-03")))
     )]
-    pub(crate) fn with_dispatch<U: Sized>(&self, inner: U) -> WithDispatch<U> {
+    pub fn with_dispatch<U: Sized>(&self, inner: U) -> WithDispatch<U> {
```

Comment:
If we're making this public, I wonder if we should also add a `U: WithSubscriber` bound? That way, if a user accidentally calls `with_dispatch` and passes in something of a type `U` for which `WithDispatch<U>` has no useful trait impls, the compiler will let them know.

What do you think?

Reply:
I'll look into this. Just a question in the mean time. U has a Sized bound. If I remember well that is the default for al type parameters, and so usually not written. Was this meant to be `?Sized`?

Reply:
I don't quite understand the `WithSubscriber` trait bound, as there is [a blanket impl](https://github.com/tokio-rs/tracing/blob/925a6980b52f918bdbd71ba5c5cc821b30cc8a1c/tracing-futures/src/lib.rs#L393) for `WithSubscriber`. It's conditional under the `std` feature.

Reply:
Ah, you're right, I forgot the impl of `WithSubscriber` was a blanket impl. Nevermind! I think the `U: Sized` is unnecessary and can be removed, but I don't remember if there was any backstory for this.